### PR TITLE
Added pagination to algorithms, archives and readerstudies list view

### PR DESCRIPTION
**Pull request by amickan**

### CODE REVIEW

File: app/grandchallenge/algorithms/views.py [hunk 1]
```diff
@@ -111,12 +112,49 @@ class AlgorithmList(FilterMixin, PermissionListMixin, ListView):
...
+
+        num_pages = int_paginator.num_pages
+        num_results = int_paginator.count
+
+        try:
+            int_page = int_paginator.page(self._current_page)
+        except EmptyPage:
+            int_page = []
+
+        return int_page, num_pages, num_results, total_count
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/views.py:141]: This seems to be more or less the same as the code added to the reader study list. Could this be consolidated into either an existing or a new mixin?

File: app/grandchallenge/algorithms/views.py [hunk 2]
```diff
@@ -111,12 +112,49 @@ class AlgorithmList(FilterMixin, PermissionListMixin, ListView):
...
 
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(*args, **kwargs)
+        page_obj, num_pages, num_results, total_count = self._get_page()
 
         context.update(
             {
+                "filter": self.int_filter,
+                "filters_applied": self._filters_applied,
```
Comment (jmsmkn) [inline app/grandchallenge/algorithms/views.py:150]: Isn't some of this functionality already in `FilterMixin`?

File: app/grandchallenge/algorithms/views.py (hunk 1 above)
Comment (amickan) [reply app/grandchallenge/algorithms/views.py]: Yes, true that. I just added a new mixin. Will change the reader study code once this and the other PR are merged.

Comment (jmsmkn) [reply app/grandchallenge/algorithms/views.py]: And if _that's_ true, maybe go one step further and change the html to a partial that you include in each page that uses it.

**Review by jmsmkn — APPROVED** [review]
LGTM 🚀

**Pull request merged**